### PR TITLE
Lsm303dlhc userspace

### PR DESCRIPTION
**Pull request by alexandruradovici**
This adds the lsm303dlhc user space for https://github.com/tock/tock/pull/1754.

The lsm303dlhc api may be used to fine tune the sensor, getting data may be used thorught the ninedof.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (ppannuto) [inline .vscode/settings.json:1]: Is this file meant to be included here?

(I don't use vscode, but we do have tool/editor config files elsewhere in Tock, so not inherently opposed to adding, but if we want to pull in some 'sane editor defaults', probably should be its own PR so people with opinions can take note more easily)

File: examples/sensors/main.c [hunk 2]
```diff
@@ -68,7 +81,7 @@ int main(void) {
   humidity    = driver_exists(DRIVER_NUM_HUMIDITY);
   ninedof     = driver_exists(DRIVER_NUM_NINEDOF);
 
-  // Setup periodic timer to sample the sensors.
+  // // Setup periodic timer to sample the sensors.
```
Comment (ppannuto) [inline examples/sensors/main.c:84]:
```suggestion
  // Setup periodic timer to sample the sensors.
```

File: examples/sensors/main.c [hunk 3]
```diff
@@ -40,17 +42,28 @@ static void timer_fired(__attribute__ ((unused)) int arg0,
...
   if (isl29035)       printf("ISL29035:   Light Intensity: %d\n", light);
   if (tmp006)         printf("TMP006:     Temperature:     %d\n", tmp006_temp);
   if (tsl2561)        printf("TSL2561:    Light:           %d lux\n", tsl2561_lux);
   if (lps25hb)        printf("LPS25HB:    Pressure:        %d\n", lps25hb_pressure);
   if (temp)           printf("Temperature:                 %d deg C\n", temp/100);
   if (humi)           printf("Humidity:                    %u%%\n", humi/100);
-  if (ninedof)        printf("FXOS8700CQ: X:               %d\n", ninedof_x);
-  if (ninedof)        printf("FXOS8700CQ: Y:               %d\n", ninedof_y);
-  if (ninedof)        printf("FXOS8700CQ: Z:               %d\n", ninedof_z);
+  if (ninedof)   {
```
Comment (ppannuto) [inline examples/sensors/main.c:52]: Hmm.., I see why you restructured this, but I think it would be better to handle all the sensors the same way. I think this app was written with hail/imix in mind, and kind of assumes all the sensors will work, but not that we have more boards with sensors, they really all need the `if .. == TOCK_SUCCESS` treatment.

---

The other issue right now, I think the spacing / alignment on the new ones don't line up (e.g. `Gryo: ` definitely needs more spaces).

File: examples/sensors/main.c (hunk 2 above)
Comment (alexandruradovici) [reply examples/sensors/main.c]: Done

File: examples/sensors/main.c (hunk 3 above)
Comment (alexandruradovici) [reply examples/sensors/main.c:52]: I fixed the gyro spaces, I think it lines up now. If we remove the ifs, if there is no sensor present, it will just display 0s. Is that ok?

File: .vscode/settings.json (hunk 1 above)
Comment (alexandruradovici) [reply .vscode/settings.json]: No, my mistake. I deleted it and added .vscode to gitignore, may others use vscode.

File: examples/sensors/main.c (hunk 3 above)
Comment (ppannuto) [reply examples/sensors/main.c:52]: I think my question is actually the opposite: should we add more if's, i.e. is the light sensor line just printing 0's on the STM board now?

Comment (alexandruradovici) [reply examples/sensors/main.c:52]: It is not printing anything, as there is no light sensor capsule (or actual sensor). It does not pass the "capsule present" if.

Comment (ppannuto) [reply examples/sensors/main.c:52]: OH. Right, I remember how this app works. Sorry.

However, with that in mind, should we expand to `ninedof_accel` `ninedof_mag` and `ninedof_gyro` booleans to match the other sensors?
